microscope/monitor/monitor.py

```diff
@@ -94,29 +97,84 @@ def connect(self):
 
         signal.signal(signal.SIGINT, sigint_in_monitor)
 
+        processor = MonitorOutputProcessor(self.mode)
+
         while resp.is_open():
+            for msg in processor:
+                if msg:
+                    self.queue.put({
+                        'name': self.pod_name,
+                        'node_name': self.node_name,
+                        'output': msg})
+
             resp.update(timeout=1)
             if not self.close_queue.empty():
                 print("Closing monitor")
                 resp.write_stdin('\x03')
                 break
             if resp.peek_stdout():
-                self.queue.put({
-                    'name': self.pod_name,
-                    'node_name': self.node_name,
-                    'output': resp.read_stdout()})
+                processor.add_out(resp.read_stdout())
             if resp.peek_stderr():
-                self.queue.put({
-                    'name': self.pod_name,
-                    'node_name': self.node_name,
-                    'output': resp.read_stderr()})
+                processor.add_err(resp.read_stderr())
 
         resp.close()
 
         self.close_queue.cancel_join_thread()
         self.queue.cancel_join_thread()
 
 
+class MonitorOutputProcessor:
+    def __init__(self, mode: str):
+        self.std_output = queuemodule.Queue()
+        self.std_err = queuemodule.Queue()
+        self.current_msg = ""
+        self.mode = mode
```

Comment:
This might be better named `verbose` or maybe a comment. I actually thought it was the line buffering mode

Reply:
Removed